File: core/core-domain/src/main/java/com/coffee/gu/order/OrderSummary.java
package com.coffee.gu.order;

import com.coffee.gu.Principal;
import com.coffee.gu.enums.OrderState;

import java.math.BigDecimal;

public record OrderSummary(
        String key,
        String name,
        Principal principal,
        BigDecimal totalPrice,
        OrderState state
) {
}

File: core/core-domain/src/main/kotlin/com/coffee/gu/order/OrderSummary.kt
package com.coffee.gu.order

import com.coffee.gu.Principal
import com.coffee.gu.enums.OrderState
import java.math.BigDecimal

data class OrderSummary(
    val key: String,
    val name: String,
    val principal: Principal,
    val totalPrice: BigDecimal,
    val state: OrderState,
)